modules/core/src/utils/logtagconfigparser.cpp

```diff
@@ -0,0 +1,218 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include "logtagconfigparser.hpp"
+
+namespace cv {
+namespace utils {
+namespace logging {
+
+LogTagConfigParser::LogTagConfigParser()
+{
+}
+
+LogTagConfigParser::LogTagConfigParser(const std::string& input)
+{
+	parse(input);
+}
+
+LogTagConfigParser::~LogTagConfigParser()
+{
+}
+
+bool LogTagConfigParser::parse(const std::string& input)
+{
+	m_input = input;
+	segmentTokens();
+	return (m_malformed.empty());
+}
+
+bool LogTagConfigParser::hasMalformed() const
+{
+    return !m_malformed.empty();
+}
+
+void LogTagConfigParser::forEachParsed(std::function<void(const std::string&, LogLevel)> func) const
+{
+    for (const auto& parsed : m_parsed)
+    {
+        func(parsed.name, parsed.level);
+    }
+}
+
+void LogTagConfigParser::forEachMalformed(std::function<void(const std::string&)> func) const
+{
+    for (const auto& malformed : m_malformed)
+    {
+        func(malformed);
+    }
+}
+
+void LogTagConfigParser::segmentTokens()
+{
+	const size_t npos = std::string::npos;
```

Comment:
This will be fixed in the next commit.
```
/build/precommit_linux64_no_opt/opencv/modules/core/src/utils/logtagconfigparser.cpp: In member function 'void cv::utils::logging::LogTagConfigParser::segmentTokens()':
/build/precommit_linux64_no_opt/opencv/modules/core/src/utils/logtagconfigparser.cpp:55:15: warning: unused variable 'npos' [-Wunused-variable]
  const size_t npos = std::string::npos;
               ^
```